esmvaltool/diag_scripts/shared/plot/styles_python/convert_ncl_style.py

```diff
@@ -32,9 +34,86 @@
 MARK = 'mark'
 AVG_STD = 'avgstd'
 FILLING = 'facecolor'
+
 INFORMATION = [DATASET, COLOR, DASH, THICKNESS, MARK, AVG_STD]
 
 
+def read_line(line):
+    """Read line of the ncl style file."""
+    # Read information
+    info_dict = {}
+    for (idx, line_elem) in enumerate(line):
+        info = line_elem.strip()
+        option = INFORMATION[idx]
+
+        # Convert color to hex string
+        if option == COLOR:
+            color = info.split(',')
+            info = '#'
+            for col in color:
+                col = "{:02x}".format(int(col))
+                info += col
+
+        # Convert mark index to matplotlib marker
+        elif option == MARK:
+            # Filling
+            if info == '16':
+                info_dict.update({FILLING: info_dict[COLOR]})
+            else:
+                info_dict.update({FILLING: 'none'})
+
```

Comment:
maybe simplify to `info_dict['FILLING'] = info_dict['COLOR'] if info == '16' else 'none'` if that fits on a single line? Why use the update method for updating a single key?